docs/source/write_first_app.rst

```diff
@@ -101,889 +97,485 @@ bringing up a new one.
-                AppraisedValue: 700,
-            },
-            {
-                ID: 'asset6',
-                Color: 'white',
-                Size: 15,
-                Owner: 'Michel',
-                AppraisedValue: 800,
-            },
-        ];
-
-        for (const asset of assets) {
-            asset.docType = 'asset';
-            await ctx.stub.putState(asset.ID, Buffer.from(JSON.stringify(asset)));
-            console.info(`Asset ${asset.ID} initialized`);
-        }
-    }
 
-The terminal output entry should look similar to below:
+An initialization function to set up initial ledger state would typically be called only once after a smart contract
+is deployed for the first time. A new version of the smart contract would likely not require the initialization to be
+performed again. In this case it is good practice for the transaction function to check and perhaps also record some
+ledger state on invocation to ensure initialization is not accidentally carried out more than once.
 
-.. code:: bash
-
-  Submit Transaction: InitLedger, function creates the initial set of assets on the ledger
-
-
-Fifth, the application invokes each of the chaincode functions
---------------------------------------------------------------
+Fifth, the application invokes transaction functions to query and modify assets
+-------------------------------------------------------------------------------
 
 First, a word about querying the ledger.
 
-Each peer in a blockchain network hosts a copy of the `ledger <./ledger/ledger.html>`_. An application
-program can view the most recent data from the ledger using read-only invocations of
-a smart contract running on your peers called a query.
+Each peer in a blockchain network hosts a copy of the `ledger <./ledger/ledger.html>`_. An application can query the
+most recent data from the ledger using read-only invocations of a smart contract running on your peers.
 
 Here is a simplified representation of how a query works:
 
 .. image:: tutorial/write_first_app.diagram.1.png
 
-The most common queries involve the current values of data in the ledger -- its
-`world state <./ledger/ledger.html#world-state>`_. The world state is
-represented as a set of key-value pairs, and applications can query data for a
-single key or multiple keys. Moreover, you can use complex queries to read the
-data on the ledger when you use CouchDB as your state database and model your data in JSON.
-This can be very helpful when looking for all assets that match certain keywords
-with particular values; all assets with a particular owner, for example.
+The application uses ``evaluateTransaction()`` to query the ledger by performing a read-only transaction function
+invocation. Behind the scenes, ``evaluateTransaction()`` will use the Gateway peer to execute the transaction function
+on a suitable peer and return its result. No transaction is sent to the ordering service and no ledger update occurs.
```

Comment:
what makes a peer suitable, we could state that instead.

Reply:
We really shouldn't. That's an implementation detail that is intended to be opaque to the client and may change over time. Instead perhaps I'll just list the key steps with less detail, since the detail is covered elsewhere.